consensus/service/src/api/client_api_service.rs

```diff
@@ -218,6 +221,24 @@ impl ConsensusClientApi for ClientApiService {
     ) {
         let _timer = SVC_COUNTERS.req(&ctx);
 
+        let session = ClientSession::from(msg.channel_id.clone());
+        match self.tracked_sessions.lock() {
+            Ok(mut tracker) => {
+                if let Some(_session_info) = tracker.get(&session) {
+                    // TODO: Update fields
+                } else {
+                    // TODO: Populate new session metadata
+                    tracker.put(session, ClientSessionTracking {});
+                }
+            }
+            Err(err) => {
+                warn!(
+                    self.logger,
+                    "Unable to acquire mutex on session metadata: {err:?}"
+                );
+            }
+        }
+
```

Comment:
:wrench: Most places with a mutex use `expect()` 
```suggestion
         {
            let mut tracker = self.tracked_sessions.lock().expect("Mutex poisoned");
            if let Some(_session_info) = tracker.get(&session) {
                // TODO: Update fields
            } else {
                // TODO: Populate new session metadata
                tracker.put(session, ClientSessionTracking {});
            }
        }
```